index.js

```diff
@@ -1,141 +1,77 @@
-var fs = require('fs');
-
-var jade           = require('jade');
-var through        = require('through');
-var transformTools = require('browserify-transform-tools');
-
-var SourceMapGenerator = require('source-map').SourceMapGenerator;
-var convert   = require('convert-source-map');
-
-var PREFIX = "var jade = require('jade/lib/runtime.js');\nmodule.exports=";
+'use strict'
+
+const pug = require('pug')
+const xtend = require('xtend')
+const through = require('through')
+const convert = require('convert-source-map')
+const tools = require('browserify-transform-tools')
+const SourceMapGenerator = require('source-map').SourceMapGenerator
+
+const defaultOptions = {
+  pretty: false,
+  compileDebug: true
+}
 
-var defaultJadeOptions = {
-  path: __dirname,
-  compileDebug: true,
-  pretty: true,
-};
+// RegEx
+const FILE_EXTENSION_REGEX = /\.(pug|jade)$/
+const LINE_NUMBER_REGEX = /^;jade_debug_line = ([0-9]+)/
 
-function getTransformFn(options) {
-  var key;
-  var opts = {};
-  for(key in defaultJadeOptions) {
-    opts[key] = defaultJadeOptions[key];
-  }
+function generateSourceMaps (src, compiled, file) {
+  const lines = compiled.split('\n')
+  const generator = new SourceMapGenerator({ file: `${file}.js` })
```

Comment:
I know it was in the old version also, but why do we need to add `.js` to the filename?